### PR TITLE
Implemented  shortcut to automatically bind scenarios to tests.

### LINKED ISSUE #103: Auto-create scenario test functions
pytest-bdd would be much more awesome if it would auto create scenario functions, so you wouldn't have to define empty functions just to make the test go. I think this either could be implemented as a test collector (also would make the test reporter more user friendly as it would show the feature file as the source of tests) or it could be just a invocation form of scenario function like:

scenario('feature/something.feature', auto=True)

### PR DESCRIPTION
**Pull request by bubenkoff**
closes #103, #89, #92, #90

### CODE REVIEW

File: pytest_bdd/scenario.py [hunk 1]
```diff
@@ -363,3 +368,40 @@ def scenario(feature_name, scenario_name, encoding="utf-8", example_converters=N
...
+        (attr.__scenario__.feature.filename, attr.__scenario__.name)
+        for name, attr in module.__dict__.items() if hasattr(attr, '__scenario__'))
+    for feature in get_features(abs_feature_paths):
+        for scenario_name, scenario_object in feature.scenarios.items():
+            # skip already bound scenarios
+            if (scenario_object.feature.filename, scenario_name) not in module_scenarios:
+                @scenario(feature.filename, scenario_name)
+                def _scenario():
+                    pass
+                module.__dict__['test_' + make_python_name(scenario_name)] = _scenario
```
Comment (olegpidsadnyi) [inline pytest_bdd/scenario.py]: incremental index in case already exists

File: tests/feature/test_scenarios.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+def test_scenarios(testdir):
+    """Test scenarios shortcut."""
+    testdir.makeconftest("""
+        import pytest
+        from pytest_bdd import given
+
+        @given('I have a bar')
+        def i_have_bar():
+            print('bar!')
```
Comment (olegpidsadnyi) [inline tests/feature/test_scenarios.py]: assert?

Comment (blueyed) [inline tests/feature/test_scenarios.py]: This seems to be done via the outer test (looking at the output).
Or is there something specific to assert in the inner test?

Comment (bubenkoff) [inline tests/feature/test_scenarios.py]: i missed the assert of the printed output - just forgot

On 18 March 2015 at 16:07, Daniel Hahler <notifications@github.com> wrote:

> In tests/feature/test_scenarios.py
> <https://github.com/pytest-dev/pytest-bdd/pull/106#discussion_r26671227>:
>
> > @@ -0,0 +1,59 @@
> > +"""Test scenarios shortcut."""
> > +import textwrap
> > +
> > +
> > +def test_scenarios(testdir):
> > +    """Test scenarios shortcut."""
> > +    testdir.makeconftest("""
> > +        import pytest
> > +        from pytest_bdd import given
> > +
> > +        @given('I have a bar')
> > +        def i_have_bar():
> > +            print('bar!')
>
> This seems to be done via the outer test (looking at the output).
> Or is there something specific to assert in the inner test?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pytest-dev/pytest-bdd/pull/106/files#r26671227>.
>



-- 
Anatoly Bubenkov

Comment (bubenkoff) [inline tests/feature/test_scenarios.py]: done

File: pytest_bdd/scenario.py (hunk 1 above)
Comment (bubenkoff) [inline pytest_bdd/scenario.py]: done

**Pull request merged**